### ISSUE TITLE
bug: core package `getNetwork` and `switchNetwork` not working with WalletConnect

### ISSUE THREAD

**Issue by zdenham** [wagmi-dev/references#393]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

"@wagmi/core" "1.3.0"

### Current Behavior

```javascript
const chains = [goerli]
    const { provider: wagmiProvider } = configureChains(chains, [
      infuraProvider({
        apiKey: 'redacted'
      }),
      w3mProvider({ projectId: this.walletConnectProjectId })
    ])

    const wagmiClient = createClient({
      autoConnect: true,
      connectors: w3mConnectors({
        projectId: this.walletConnectProjectId,
        version: 2,
        chains
      }),
      provider: wagmiProvider,
      chains
    })
 ```
 
- Connect wallet by scanning wallet connect QR code with mobile metamask
- Metamask on mobile is on mainnet chainId 1 when connecting
- Call `getNetwork` from wagmi/core returns the incorrect chainId 5 (goerli)
- Call `switchNetwork` does nothing

### Expected Behavior

- expect call to `getNetwork` to return the chain which is currently connected (on mobile metamask)
- expect call to `switchNetwork` to prompt the user on mobile wallet to switch chains

### Steps To Reproduce

See above

### Anything else?

We are using the w3mConnectors with infura fallback provider for reads

**Comment by siosio34** [comment]
@zdenham 
Have you ever solved this error?

Did the code below help?
```tsx
const wagmiClient = createConfig({
      autoConnect: true,
      connectors: [ w3mConnectors({
        projectId: this.walletConnectProjectId,
        version: 2,
        chains
      })],
      publicClient,
      webSocketPublicClient,
    })
```

**Comment by zdenham** [comment]
> Did the code below help?

This code is from wagmi ^1.0 which has viem peer dependency, which for our project is not feasible to switch to in the near term

**Comment by jingxuan98** [comment]
still having this issue any fix? 

When calling the switchNetwork() in wagmi, the walletClient, signer and network on my dapp switched to the correct network but not my metamask mobile (no-popup and stay at the previous network page), then when try to proceed with a contract interaction theres no pop-up as well, but everything works fine after switching to the correct network manually in metamask mobile

**Comment by zdenham** [comment]
I have a feeling the issue is either w/ wallet connect connector or MM mobile itself. But no resolution so far for me

**Comment by siosio34** [comment]
@luisdeka If I apply this code, can I change the chain within WalletConnect?

**Comment by luisdeka** [comment]
> @luisdeka If I apply this code, can I change the chain within WalletConnect?

for what i have test, yes

**Comment by siosio34** [comment]
I'm really sorry but can you show me the whole code from connection to network change?

I don't know if I understood what you said correctly, but from what I understand, it is as follows.

```tsx
// This solves the problem that the network switching popup does not appear in the app.
setTimeout(() => connect({ chainId })), 1500)

// In this way, after performing the switchNetwork process, the next process did not proceed.
setTimeout(() => switchNetwork({ chainId }).then(checkIfChangedAndContinueProcress), 3500)
```

Is my understanding of the above code correct?

**Comment by luisdeka** [comment]
> thanks @luisdeka  I'm really sorry but can you show me the whole code from connection to network change?
> 
> I don't know if I understood what you said correctly, but from what I understand, it is as follows.
> 
> ```tsx
> // This solves the problem that the network switching popup does not appear in the app.
> setTimeout(() => connect({ chainId })), 1500)
> 
> // In this way, after performing the switchNetwork process, the next process did not proceed.
> setTimeout(() => switchNetwork({ chainId }).then(checkIfChangedAndContinueProcress), 3500)
> ```
> 
> Is my understanding of the above code correct?
> 
> 

I have not tested the first one, the second is exactly that.

**Comment by lunarinlunarinlunarin** [comment]
is there an update on this? same issue for me

**Comment by imaksp** [comment]
> is there an update on this? same issue for me

see this issue, it will be fixed in MM 7.4 version.
https://github.com/MetaMask/metamask-mobile/issues/6655#issuecomment-1658587787
